### DISCUSSION TITLE
Libbacktrace and back to upstream question

### DISCUSSION THREAD

**Discussion by gvozdila** [bloomberg/memray#64] [Q&A] [answered]
Hello! I'm going to be AltLinux maintainer for memray and i have a question about libbacktrace.
Looks like you apply some patch on libbacktrace before using it.
Have you consider sharing you changes with libbacktrace upstream?

**Comment by pablogsal** [comment, marked as answer]
Yes, but we don't really want to do this for the time being. The patch in question exposes a bunch of internal functions so we can use them but we may need to do other changes that are `memray` exclusive in the future. For this reason we don't want to push (for the time being) upstream maintainers to make public the interfaces we need because we may need even more aggressive changes in the future and we still want users to be able to build `memray` if their distribution has an old version of `libbacktrace`.

If the situation becomes stable and we see that we are unlikely going to further modify the vendored lib, we may consider proposing upstreaming those changes.

**Reply by gvozdila** [reply]
Ok, i got it. Thank you for your work.

**Reply by pablogsal** [reply]
Thanks for checking with us ♥️
